utils/frame/benchmarking-cli/src/writer.rs

```diff
@@ -130,57 +95,102 @@ pub fn write_results(file: &mut File, batches: Vec<BenchmarkBatch>) -> Result<()
 
 		// only create new trait definitions when we go to a new pallet
 		if batch.pallet != current_pallet {
-			if !current_pallet.is_empty() {
-				// close trait
-				write!(file, "}}\n").unwrap();
-			}
+			// auto-generation note
+			write!(
+				file,
+				"//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION {}\n\n",
+				 VERSION,
+			).unwrap();
```

Comment:
I'm not sure why you unwrap everything when the function result with std::io::Error

Reply:
I have fixed a bunch here, but there are some error types that dont convert... so left those